packages/common/src/hooks/useTrackDogEar.tsx

```diff
@@ -1,39 +1,41 @@
 import { useSelector } from 'react-redux'
 
-import { useGetTrackById } from '~/api'
 import { DogEarType } from '~/models'
 import { ID } from '~/models/Identifiers'
 import {
   isContentCollectibleGated,
   isContentSpecialAccess,
-  isContentUSDCPurchaseGated,
-  Track
+  isContentUSDCPurchaseGated
 } from '~/models/Track'
+import { CommonState } from '~/store'
 import { getUserId } from '~/store/account/selectors'
+import { getTrack } from '~/store/cache/tracks/selectors'
 import { Nullable } from '~/utils'
 
-import { useGatedContentAccess } from './useGatedContent'
+import { useGatedTrackAccess } from './useGatedContent'
 
 export const useTrackDogEar = (trackId: ID, hideUnlocked = false) => {
-  const { data: track } = useGetTrackById({ id: trackId })
-  const currentUserId = useSelector(getUserId)
+  const streamConditions = useSelector((state: CommonState) => {
+    return getTrack(state, { id: trackId })?.stream_conditions
+  })
 
-  const { hasStreamAccess, hasDownloadAccess } = useGatedContentAccess(
-    track as Nullable<Track>
-  )
+  const downloadConditions = useSelector((state: CommonState) => {
+    return getTrack(state, { id: trackId })?.download_conditions
+  })
 
-  if (!track) return null
+  const isOwner = useSelector((state: CommonState) => {
```

Comment:
seen this selector a few times, maybe pull out to shared hook somewhere?

Reply:
yeah i agree! when we do react-query lets def make sure that happens